### PR TITLE
fix: hide indicator outside the droppable area (#477)

### LINKED ISSUE #477: The indicator and highlight is still active when out droppable area
https://github.com/ant-design/ant-design/issues/31555

https://user-images.githubusercontent.com/76864176/127765439-3bfa51ee-03fb-4b4f-8f59-0bc4335e27ea.mov

### PR DESCRIPTION
**Pull request by curly210102**
Resolves ant-design/ant-design#31555, resolves #477

### CODE REVIEW

File: src/Tree.tsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ import {
   calcDropPosition,
   arrAdd,
   arrDel,
-  posToArr,
+  posToArr
```
Comment (07akioni) [inline src/Tree.tsx:25]:
```suggestion
  posToArr,
```

File: src/Tree.tsx [hunk 2]
```diff
@@ -278,6 +278,8 @@ class Tree extends React.Component<TreeProps, TreeState> {
 
   dragNode: NodeInstance;
 
+  dragEnterDroppableNodeKey = null;
```
Comment (07akioni) [inline src/Tree.tsx:281]: currentMouseOverDroppableNodeKey

**Comment by curly210102** [comment]
@07akioni find a new problem with this solution, drag enter/leave event fired when hovering children element. Refer to https://stackoverflow.com/questions/7110353/html5-dragleave-fired-when-hovering-a-child-element, I add a dragenter counter to check whether the mouse leave the draggable element.

**Comment by 07akioni** [comment]
> , drag enter/leave event fired when hovering children element. R

Why not check currentTarget?

**Comment by curly210102** [comment]
Checking event.currentTarget === event.target? As the cursor switches between the internal blank area and the child element, event.target will return the draggable element node multiple times([Codepen Demo](https://codepen.io/curly210102/pen/qBjEjjp)). Thus, It cannot used to check the cursor has left.

**Pull request merged**